pkg/services/pluginsintegration/serviceregistration/serviceregistration.go

```diff
@@ -58,35 +58,16 @@ func (s *Service) RegisterExternalService(ctx context.Context, pluginID string,
 
 		enabled = (settings != nil) && settings.Enabled
 	}
-
-	impersonation := extsvcauth.ImpersonationCfg{}
-	if svc.Impersonation != nil {
-		impersonation.Permissions = toAccessControlPermissions(svc.Impersonation.Permissions)
-		impersonation.Enabled = enabled
-		if svc.Impersonation.Groups != nil {
-			impersonation.Groups = *svc.Impersonation.Groups
-		} else {
-			impersonation.Groups = true
-		}
-	}
-
 	self := extsvcauth.SelfCfg{}
 	self.Enabled = enabled
 	if len(svc.Permissions) > 0 {
 		self.Permissions = toAccessControlPermissions(svc.Permissions)
 	}
 
 	registration := &extsvcauth.ExternalServiceRegistration{
-		Name:          pluginID,
-		Impersonation: impersonation,
-		Self:          self,
-	}
-
-	// Default authProvider now is ServiceAccounts
-	registration.AuthProvider = extsvcauth.ServiceAccounts
-	if svc.Impersonation != nil {
-		registration.AuthProvider = extsvcauth.OAuth2Server
-		registration.OAuthProviderCfg = &extsvcauth.OAuthProviderCfg{Key: &extsvcauth.KeyOption{Generate: true}}
+		Name:         pluginID,
+		Self:         self,
+		AuthProvider: extsvcauth.ServiceAccounts,
```

Comment:
same: AuthProvider is going to be always `ServiceAccounts`?

Reply:
Yes currently we'd only have this provider given I'm removing the other one.
I'll track this potential clean up in a separate issue